java/memory/src/main/java/org/apache/arrow/memory/NettyAllocationManager.java

```diff
@@ -34,31 +33,24 @@
   static final UnsafeDirectLittleEndian EMPTY = INNER_ALLOCATOR.empty;
   static final long CHUNK_SIZE = INNER_ALLOCATOR.getChunkSize();
 
-  private final int allocatedSize;
-  private final UnsafeDirectLittleEndian memoryChunk;
+  private final long allocatedSize;
 
-  NettyAllocationManager(BaseAllocator accountingAllocator, int requestedSize) {
-    super(accountingAllocator);
-    this.memoryChunk = INNER_ALLOCATOR.allocate(requestedSize);
```

Comment:
I don't think we should remove this which effectively replaces all allocations done in Arrow Java, which is a big change. `INNER_ALLOCATOR` also uses a pool which has some benefits. Instead, can you just change `requestedSize` to be a long, then check if it is over the max Int size and only then use `PlatformDependent.allocateMemory`?

Reply:
Revised. Thank you for the good suggestion.